index.js

```diff
@@ -82,7 +82,7 @@ module.exports = function (file, opts) {
         
         varNames.forEach(function (name) {
             if (scope.globals.implicit.indexOf(name) >= 0) {
-                var value = vars[name](file, basedir);
+                var value = vars[name] && vars[name](file, basedir);
```

Comment:
I like the tests and idea, but I don't think this belongs here. If instead you did:

```diff
diff --git a/index.js b/index.js
index 7780505e..65f62b9e 100644
--- a/index.js
+++ b/index.js
@@ -35,7 +35,9 @@ module.exports = function (file, opts) {
     
     var basedir = opts.basedir || '/';
     var vars = merge(defaultVars, opts.vars);
-    var varNames = Object.keys(vars);
+    var varNames = Object.keys(vars).filter(function(name) {
+        return typeof vars[name] === 'function';
+    });
     
     var quick = RegExp(varNames.map(function (name) {
         return '\\b' + name + '\\b';
@@ -82,7 +84,7 @@ module.exports = function (file, opts) {
         
         varNames.forEach(function (name) {
             if (scope.globals.implicit.indexOf(name) >= 0) {
-                var value = vars[name] && vars[name](file, basedir);
+                var value = vars[name](file, basedir);
                 if (value) {
                     globals[name] = value;
                     self.emit('global', name);
```

Then you'd also get the benefit of the "quick test", to avoid parsing files unnecessarily.